File: pawfinder_app_frontend/lib/domain/entities/pet.dart
import 'package:equatable/equatable.dart';

/// Domain entity representing a registered pet.
class Pet extends Equatable {
  final String id;
  final String ownerId;
  final String name;
  final String species;
  final String? breed;
  final String? color;
  final String? size;
  final String? distinguishingFeatures;
  final String? temperamentNotes;
  final String? medicalNeeds;
  final List<String> photos;
  final String? qrCodeId;
  final DateTime createdAt;

  const Pet({
    required this.id,
    required this.ownerId,
    required this.name,
    required this.species,
    this.breed,
    this.color,
    this.size,
    this.distinguishingFeatures,
    this.temperamentNotes,
    this.medicalNeeds,
    this.photos = const [],
    this.qrCodeId,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        ownerId,
        name,
        species,
        breed,
        color,
        size,
        distinguishingFeatures,
        temperamentNotes,
        medicalNeeds,
        photos,
        qrCodeId,
        createdAt,
      ];
}

File: pawfinder-app-backend/pawfinder-shared/src/main/java/com/pawfinder/shared/domain/Pet.java
package com.pawfinder.shared.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "pets")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "owner_id", insertable = false, updatable = false)
    private UUID ownerId;

    private String name;

    private String species;

    private String breed;

    private String color;

    private String size;

    @Column(name = "distinguishing_features", columnDefinition = "TEXT")
    private String distinguishingFeatures;

    @Column(name = "temperament_notes", columnDefinition = "TEXT")
    private String temperamentNotes;

    @Column(name = "medical_needs", columnDefinition = "TEXT")
    private String medicalNeeds;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "JSONB")
    private String photos;

    @Column(name = "qr_code_id", unique = true)
    private String qrCodeId;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}
